python/ray/util/ray_executor.py

```diff
@@ -0,0 +1,277 @@
+                future = ex.submit(lambda x: x * x, 100)
+                result = future.result()
+        """
+        self._check_shutdown_lock()
+        oref = self.__remote_fn.remote(fn, *args, **kwargs)
+        future = oref.future()
+        self._futures[oref] = future
+        return future
+
+    def submit_actor_function(self, fn, *args, **kwargs):
+        """Submits a callable Actor function (see
+        https://docs.ray.io/en/latest/ray-core/actors.html) to be executed with
+        the given arguments.
+
+        Schedules the callable to be executed as `fn(*args, **kwargs)` and returns
+        a Future instance representing the execution of the callable.
+
+        Returns:
+            A Future representing the given call.
+
+
+        Usage example:
+
+        .. code-block:: python
+
+            @ray.remote
+            class Actor0:
+                def __init__(self, name):
+                    self.name = name
+
+                def actor_function(self, arg):
+                    return f"{self.name}-Actor-{arg}"
+
+            a = Actor0.options(name="A", get_if_exists=True).remote("A")
+            with RayExecutor() as ex:
+                future = ex.submit_actor_function(a.actor_function, 0)
+                result = future.result()
+
+        """
+        self._check_shutdown_lock()
+        oref = self.__actor_fn(fn, *args, **kwargs)
+        future = oref.future()
+        self._futures[oref] = future
+        return future
+
+    def map(self, fn, *iterables, timeout=None, chunksize=1):
+        """Returns an iterator equivalent to `map(fn, iter)`.
+
+        Args:
+            fn: A callable that will take as many arguments as there are
+                passed iterables.
+            timeout: The maximum number of seconds to wait. If None, then there
+                is no limit on the wait time.
+            chunksize: The size of the chunks the iterable will be broken into
+                before being passed to a child process.
+
+        Returns:
+            An iterator equivalent to: `map(func, *iterables)` but the calls may
+            be evaluated out-of-order.
+
+        Raises:
+            TimeoutError: If the entire result iterator could not be generated
+                before the given timeout.
+            Exception: If `fn(*args)` raises for any values.
+
+        Usage example:
+
+        .. code-block:: python
+
+            with RayExecutor() as ex:
+                futures = ex.map(lambda x: x * x, [100, 100, 100])
+                results = [future.result() for future in futures()]
+
+        """
+        self._check_shutdown_lock()
+        results_list = []
+        for chunk in self._get_chunks(*iterables, chunksize=chunksize):
+            results = self._map(self.submit, fn, chunk, timeout)
+            results_list.append(results)
+        return itertools.chain(*results_list)
+
+    def map_actor_function(self, fn, *iterables, timeout=None, chunksize=1):
```

Comment:
should this be a private method? `_ map_actor_function `

Reply:
No, this method is the one to call when someone wants to use the `RayExecutor` with an actor function.